### ISSUE TITLE
Rule Request: Unreferenced Files

### ISSUE THREAD

**Issue by mohamede1945** [realm/SwiftLint#2559]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Sometimes people remove a reference to a swift file from an Xcode project or a target but forgot to delete the file. Hence the file remains in the the source control and nobody understands this file is not used at all.

1. Why should this rule be added? 
Usually iOS developers look at Xcode and rarely look at the Finder. If there is a difference like there is a file that is no longer used it would set in the source control unreferenced. Removing such files would be beneficial to making the repo tight.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.
Trigger:
```swift
Xcode Project: {}
Foo.swift // Please delete any unreferenced files from disk. (unreferenced_file)
```

Doesn't trigger:
```swift
Xcode Project: {
    somewhere: Foo.swift
}
Foo.swift
```

3. Should the rule be configurable, if so what parameters should be configurable?
Yes, specifying Xcode project file or workspace.

4. Should the rule be opt-in or enabled by default? Why?
Yes, some repos might not need xcodeproject at all for example swift scripting.

**Comment by Igor-Palaguta** [comment]
There is a tool that should lint xcode project https://github.com/americanexpress/xcprojectlint

**Comment by jpsim** [comment]
I agree that this is a better fit for a tool like an Xcode project linter. SwiftLint has really just been focused on linting only swift files.
